packages/perf/perftool-testsuite/runtest.sh

```diff
@@ -0,0 +1,293 @@
+		if [[ $ARCH =~ ppc64.* ]]; then
+			# detect POWER virtualization
+			rlLog "Virtualization: `systemd-detect-virt -q && echo PowerKVM || ( test -e /proc/ppc64/lparcfg && echo PowerVM || echo none )`"
+		else
+			# detect virtualization
+			rlLog "Virtualization: `virt-what`"
+		fi
+
+		export KERNEL_DEBUGINFO_PKG_NAME="kernel-debuginfo-$KERNEL"
+		if [ $(is_kernel_rt) -eq 0 ]; then
+			export KERNEL_DEBUGINFO_PKG_NAME="kernel-rt-debuginfo-$KERNEL"
+		fi
+		export KERNEL_PKG_NAME="kernel-$KERNEL"
+		echo $KERNEL | grep -q debug
+		if [ $? -eq 0 ]; then
+			export KERNEL=${KERNEL%[.+]debug}
+			export KERNEL_PKG_NAME="kernel-debug-$KERNEL"
+			export KERNEL_DEBUGINFO_PKG_NAME="kernel-debug-debuginfo-$KERNEL"
+			if [ $(is_kernel_rt) -eq 0 ]; then
+				export KERNEL_PKG_NAME="kernel-rt-debug-$KERNEL"
+				export KERNEL_DEBUGINFO_PKG_NAME="kernel-rt-debug-debuginfo-$KERNEL"
+			fi
+		fi
+
+		rlLog "Variables:"
+		rlLog "KERNEL = $KERNEL"
+		rlLog "KERNEL_PKG_NAME = $KERNEL_PKG_NAME"
+		rlLog "KERNEL_DEBUGINFO_PKG_NAME = $KERNEL_DEBUGINFO_PKG_NAME"
+		rpmquery $KERNEL_DEBUGINFO_PKG_NAME
+		if [ $? -ne 0 ]; then
+			# we need to install debuginfo for the proper kernel
+			# but sometimes, debuginfo-install is not available!
+			which debuginfo-install || rlRun "yum -y install yum-utils dnf-utils" 0 "Installing {yum,dnf}-utils (it has not been present)"
+			which debuginfo-install # now it should be installed, but what if it fails...
+			if [ $? -eq 0 ]; then
+				rlRun "debuginfo-install -y $KERNEL_PKG_NAME" 0 "Installing debuginfo for $KERNEL_PKG_NAME via debuginfo-install (it has not been present)"
+			else
+				rlRun "yum install -y $KERNEL_DEBUGINFO_PKG_NAME" 0 "Installing debuginfo for $KERNEL_PKG_NAME via yum/dnf (unable to obtain debuginfo-install)"
+			fi
+		fi
+		rlRun "rpmquery $KERNEL_DEBUGINFO_PKG_NAME" 0 "Correct debuginfo is installed ($KERNEL)"
+		# return Skip when correct kernel debug is not installed
+		if [ $? -ne 0 ]; then
+			skip_testcase "Correct kernel debuginfo pkg: ${KERNEL_DEBUGINFO_PKG_NAME} is not installed"
+		fi
+		echo "==================== kernel packages installed ===================="
+		rpmquery -a | grep -e kernel -e perf
+		echo "==================================================================="
+
+		# log whether we use blacklisting
+		if [ $PERFTESTS_ENABLE_BLACKLIST -ne 0 ]; then
+			rlLog "BLACKLISTING ENABLED (known fails will be hidden)"
+		else
+			rlLog "BLACKLISTING DISABLED"
+		fi
+
+		# set kptr_restrict to 0
+		PREVIOUS_KPTR_RESTRICT=`cat /proc/sys/kernel/kptr_restrict`
+		rlRun "echo 0 > /proc/sys/kernel/kptr_restrict"
+		rlAssert0 "kptr_restrict must be set to 0" `cat /proc/sys/kernel/kptr_restrict`
+
+		# clone the upstream perftool-testsuite repo
+		fetch_the_testsuite
+		# return SKip if the testsuite could not be fetched
+		test -d "perftool-testsuite" || skip_testcase "Could not fetch the upstream testsuite from github. I am sorry, dude."
```

Comment:
Would it make more sense to abort the execution if the upstream testsuite can't be reached? Aborted tests grab attention and not being able to clone the repo might also mean that the repo was migrated or removed (extreme, but might happen) and we won't know about it as people don't usually look at the logs of skipped tests.